File: chazm-model/src/main/java/runtimemodels/chazm/model/entity/CapabilityFactory.java
package runtimemodels.chazm.model.entity;

import runtimemodels.chazm.api.entity.Capability;
import runtimemodels.chazm.api.id.UniqueId;

/**
 * The {@linkplain CapabilityFactory} interface defines the APIs for constructing {@linkplain Capability}s.
 *
 * @author Christopher Zhong
 * @since 7.0.0
 */
@FunctionalInterface
public interface CapabilityFactory {

    /**
     * Constructs a {@linkplain Capability}.
     *
     * @param id the {@linkplain UniqueId} that represents the {@linkplain Capability}.
     * @return a {@linkplain Capability}.
     */
    Capability buildCapability(UniqueId<Capability> id);

}

File: chazm-model/src/main/kotlin/runtimemodels/chazm/model/entity/CapabilityFactory.kt
package runtimemodels.chazm.model.entity

import runtimemodels.chazm.api.entity.Capability
import runtimemodels.chazm.api.entity.CapabilityId
import runtimemodels.chazm.api.id.UniqueId

/**
 * The [CapabilityFactory] interface defines the APIs for constructing [Capability]s.
 *
 * @author Christopher Zhong
 * @since 7.0.0
 */
@FunctionalInterface
interface CapabilityFactory {
    /**
     * Constructs a [Capability].
     *
     * @param id the [UniqueId] that represents the [Capability].
     * @return a [Capability].
     */
    fun build(id: CapabilityId): Capability
}
